src/container/container.ts

```diff
@@ -637,7 +637,7 @@ class Container implements interfaces.Container {
     }
 
     private _deactivateIfSingleton(binding: Binding<any>): Promise<void> | void {
```

Comment:
Maybe this method should be called `_deactivateIfActivated` instead

Reply:
It is clearer that deactivation applies to singletons only with _deactivateIfSingleton.  I think activated is a bad name choice given anything can be "activated" through onActivation.  

The [v6-binding-abstractions](https://github.com/inversify/InversifyJS/pull/1336) change I made hopefully makes it clearer.

```
    private _deactivateIfSingleton(binding: interfaces.Binding<any>): Promise<void> | void {
        const scope = binding.scope;
        if(scope.type === ConfigurableBindingScopeEnum.Singleton){
            const cached = scope.resolved;
            if (cached === undefined) {
                return;
            }

            if (isPromise(cached)) {
                return cached.then((resolved: any) => this._deactivate(binding, resolved));
            }

            return this._deactivate(binding, cached);
        }

    }

```